core_impl/src/test/java/com/google/instrumentation/stats/StatsContextTest.java

```diff
@@ -167,17 +169,21 @@ public void testToString() {
   }
 
   private static void testSerialize(Tag... tags) throws IOException {
+    final Map<TagKey, TagValue> tagMap = new HashMap<TagKey, TagValue>();
+    for (Tag tag : tags) {
+      tagMap.put(tag.getKey(), tag.getValue());
+    }
+
     ByteArrayOutputStream actual = new ByteArrayOutputStream();
+    new StatsContextImpl(tagMap).serialize(actual);
+
     ByteArrayOutputStream expected = new ByteArrayOutputStream();
     expected.write(VERSION_ID);
-    StatsContext.Builder builder = DEFAULT.builder();
-    for (Tag tag : tags) {
-      builder.set(tag.getKey(), tag.getValue());
+    for (Entry<TagKey, TagValue> entry : tagMap.entrySet()) {
       expected.write(VALUE_TYPE_STRING);
-      encodeString(tag.getKey().toString(), expected);
-      encodeString(tag.getValue().toString(), expected);
+      encodeString(entry.getKey().asString(), expected);
+      encodeString(entry.getValue().asString(), expected);
     }
-    builder.build().serialize(actual);
 
     assertThat(actual.toByteArray()).isEqualTo(expected.toByteArray());
```

Comment:
I think it would be better to test through the StatsContext API, to make the test less fragile.  You could use Guava's permutations method to generate all permutations of `tags`: https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Collections2.html#permutations(java.util.Collection)